lib/magma/attribute.rb

```diff
@@ -15,45 +34,56 @@ class Attribute
       :validation
     ]
 
-    attr_reader :name, :loader, :validation, :format_hint, :unique, :index, :restricted, :link_model_name, :description
+    attr_reader :loader
 
     class << self
       def options
         [:description, :display_name, :hidden, :read_only, :unique, :index, :validation,
 :format_hint, :loader, :link_model_name, :restricted]
       end
 
-      def set_attribute(name, model, options, attribute_class)
-        attribute_class.new(name, model, options)
-      end
-
       def attribute_type
         @attribute_type ||= name.match("Magma::(.*)Attribute")[1].underscore
       end
     end
 
-    def initialize(name, model, opts)
-      @name = name
-      @model = model
-      set_options(opts)
+    def initialize(opts = {})
+      # Some Ipi models set the group option but it doesn't seem to be used anywhere
+      opts.delete(:group)
+
+      super(opts.except(:magma_model, :loader))
+      self.magma_model = opts[:magma_model]
+      @loader = opts[:loader]
+    end
+
+    def magma_model=(new_magma_model)
+      @magma_model = new_magma_model
+      after_magma_model_set
+    end
+
+    def after_magma_model_set
+    end
+
+    def name
+      attribute_name.to_sym
```

Comment:
There are a few places in the codebase that need `attribute_name` to be a symbol, so I left this here as a convenience method. I'd be open to removing it though.